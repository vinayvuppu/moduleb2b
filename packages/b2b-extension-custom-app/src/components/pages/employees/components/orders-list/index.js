export { default } from './orders-list';
export * from './orders-list';
