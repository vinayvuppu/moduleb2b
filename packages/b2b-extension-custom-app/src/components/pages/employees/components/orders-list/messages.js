import { defineMessages } from 'react-intl';

export default defineMessages({
  columnOrderNumber: {
    id: 'Employees.Details.OrdersForm.columnOrderNumber',
    description: 'The column label (Order number)',
    defaultMessage: 'Order number',
  },
  columnCreatedAt: {
    id: 'Employees.Details.OrdersForm.columnCreatedAt',
    description: 'The column label (Created at)',
    defaultMessage: 'Created on',
  },
  columnOrderStatus: {
    id: 'Employees.Details.OrdersForm.columnOrderStatus',
    description: 'The column label (Order state)',
    defaultMessage: 'Order Status',
  },
  columnPaymentState: {
    id: 'Employees.Details.OrdersForm.paymentState',
    description: 'Title of the table column (paymentState)',
    defaultMessage: 'Payment Status',
  },
  columnShipmentState: {
    id: 'Employees.Details.OrdersForm.shipmentState',
    description: 'Title of the table column (shipmentState)',
    defaultMessage: 'Shipment Status',
  },
  columnOrderTotal: {
    id: 'Employees.Details.OrdersForm.columnOrderTotal',
    description: 'The column label (Order total)',
    defaultMessage: 'Order total',
  },

  noOrdersTitle: {
    id: 'Employees.Details.OrdersForm.noOrdersTitle',
    description: 'Title for no orders found.',
    defaultMessage: 'There are no orders related to this employee.',
  },

  fetchingOrdersTitle: {
    id: 'Employees.Details.OrdersForm.fetchingOrdersTitle',
    description: 'Title shown while fetching orders.',
    defaultMessage: 'Loading orders...',
  },
});
