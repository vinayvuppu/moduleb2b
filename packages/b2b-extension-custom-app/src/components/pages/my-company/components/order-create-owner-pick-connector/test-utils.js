/* eslint-disable import/prefer-default-export */
import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import flatMap from 'lodash.flatmap';
import intersection from 'lodash.intersection';
import FetchCustomersQuery from './order-create-customer-pick-connector.graphql';

export const createCustomersFetchQueryMock = (stores = []) => {
  const whereClause =
    stores.length === 0
      ? undefined
      : `(stores(key = "${stores.join(',')}") or stores is empty)`;
  const customers = [
    {
      id: 'customer-1',
      version: 4,
      firstName: 'Anna',
      lastName: 'Banana',
      email: 'annabanana@blablub.com',
      customerNumber: '111',
      companyName: null,
      customerGroup: null,
      stores: [
        {
          id: 'store-2',
          key: 'usa',
          nameAllLocales: [{ locale: 'en-GB', value: 'USA' }],
        },
      ],
      addresses: [
        {
          id: '_ZqTe0es',
          firstName: '3333',
          streetName: 'Fakestreet',
          streetNumber: '23',
          country: 'DE',
          additionalAddressInfo: 'Additional address info2',
          additionalStreetInfo: 'Additional street info',
          email: null,
          phone: null,
          city: null,
          company: null,
          lastName: null,
          postalCode: null,
          region: null,
        },
      ],
    },
    {
      id: 'customer-2',
      version: 1,
      firstName: 'Ali',
      lastName: 'Baba',
      customerNumber: '235',
      email: 'alibaba@testest.de',
      companyName: null,
      customerGroup: null,
      stores: [
        {
          id: 'store-1',
          key: 'germany',
          nameAllLocales: [{ locale: 'en-GB', value: 'Germany' }],
        },
      ],
      addresses: [],
    },
    {
      id: 'customer-3',
      version: 3,
      customerNumber: '128',
      firstName: 'Amin',
      lastName: 'Rocks',
      email: 'email@ct.de',
      companyName: null,
      customerGroup: null,
      stores: [],
      addresses: [
        {
          id: 'yUQGmqRG',
          firstName: 'first',
          lastName: 'last2',
          company: 'company',
          streetName: 'Street name',
          streetNumber: 'House number',
          city: 'city',
          postalCode: '66215',
          region: 'region',
          country: 'US',
          additionalAddressInfo: 'Additional address info2',
          additionalStreetInfo: 'Additional street info',
          email: 'email2@email.com',
          phone: 'phone',
        },
      ],
    },
  ].filter(customer => {
    // if we don't pass stores then we don't filter
    if (!stores) return true;

    const customerStoresArray = flatMap(customer.stores, s => s.key);
    return (
      // true if the customer belongs to one of the stores passed
      intersection(customerStoresArray, stores).length !== 0 ||
      // or the customer belongs to no store
      customerStoresArray.length === 0
    );
  });

  return {
    request: {
      variables: {
        target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
        projectKey: 'context-playground',
        limit: 20,
        locale: 'en',
        offset: 0,
        sort: ['createdAt desc'],
        where: whereClause,
      },
      query: FetchCustomersQuery,
    },
    result: {
      data: {
        customers: {
          count: 3,
          offset: 0,
          total: 3,
          results: customers,
        },
      },
    },
  };
};
