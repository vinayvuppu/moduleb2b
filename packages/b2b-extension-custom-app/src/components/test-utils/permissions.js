import { PERMISSIONS } from '../../constants';

// eslint-disable-next-line import/prefer-default-export
export const allTestPermissions = Object.keys(PERMISSIONS).reduce(
  (permissionsMap, permissionKey) => ({
    ...permissionsMap,
    [`can${permissionKey}`]: true,
  }),
  {}
);
