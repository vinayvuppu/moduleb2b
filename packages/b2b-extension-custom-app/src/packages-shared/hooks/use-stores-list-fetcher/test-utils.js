import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
import { FetchStoresQuery } from './stores-list-query.graphql';

// eslint-disable-next-line import/prefer-default-export
export const createFetchStoresListQueryMock = (custom = {}) => ({
  request: {
    variables: {
      target: GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM,
      limit: 20,
      offset: 0,
      sort: `createdAt desc`,
    },
    ...custom.request,
    query: FetchStoresQuery,
  },
  result: {
    data: {
      stores: {
        count: 2,
        offset: 0,
        total: 2,
        results: [
          {
            id: 'store-1',
            key: 'germany',
            version: 1,
            nameAllLocales: [{ locale: 'en-GB', value: 'Germany' }],
            createdAt: '2018-12-19T15:26:51.174Z',
            lastModifiedAt: '2019-03-01T09:00:51.174Z',
          },
          {
            id: 'store-2',
            key: 'usa',
            version: 1,
            nameAllLocales: [{ locale: 'en-GB', value: 'USA' }],
            createdAt: '2019-06-10T15:00:51.140Z',
            lastModifiedAt: '2019-08-30T10:26:51.140Z',
          },
        ],
      },
    },
    ...custom.result,
  },
});
