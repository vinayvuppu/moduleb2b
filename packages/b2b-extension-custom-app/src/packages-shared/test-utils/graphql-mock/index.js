export { createRosie, createGraphqlMock } from './graphql-mock';
