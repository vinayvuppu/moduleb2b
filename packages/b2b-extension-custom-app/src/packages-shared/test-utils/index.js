import crypto from 'crypto';
import { withProps, compose, branch, renderComponent } from 'recompose';

export { createGraphqlMock, createRosie } from './graphql-mock';

export const intlMock = {
  formatMessage: jest.fn(message => message.id),
  formatDate: jest.fn(date => {
    const dateObj = new Date(date);
    const day = dateObj.getUTCDate();
    const month = dateObj.getUTCMonth() + 1;
    const year = dateObj.getUTCFullYear();
    return `${day}.${month}.${year}`;
  }),
  formatTime: jest.fn(date => {
    const dateObj = new Date(date);
    const hours = dateObj.getUTCHours();
    const minutes = dateObj.getUTCMinutes();
    return `${hours}:${minutes}`;
  }),
  formatRelative: jest.fn(() => 'xxx'),
  formatNumber: jest.fn((number, currency) =>
    currency?.currency
      ? `${currency.currency} ${number.toString()}`
      : number.toString()
  ),
  formatPlural: jest.fn(() => 'xxx'),
  formatHTMLMessage: jest.fn(() => 'xxx'),
  now: jest.fn(() => 'xxx'),
  locale: 'en',
};

export const routerMock = {
  go: jest.fn(() => {}),
  goBack: jest.fn(() => {}),
  goForward: jest.fn(() => {}),
  push: jest.fn(() => {}),
  replace: jest.fn(() => {}),
  isActive: jest.fn(() => {}),
  createHref: jest.fn(() => {}),
  setRouteLeaveHook: jest.fn(() => {}),
};

export function createMountOptions(options = {}) {
  const { intl, router } = options;
  return {
    context: {
      intl: { ...intlMock, ...intl },
      router: { ...routerMock, ...router },
      store: {
        getState: () => ({
          application: {
            products: {
              currentVisible: {
                productType: {
                  obj: {
                    name: 'foo',
                  },
                },
              },
            },
          },
        }),
        subscribe() {},
        dispatch() {},
      },
    },
  };
}

export function promisedTimeout(cb, timeout = 0) {
  const p = new Promise(resolve => {
    setTimeout(() => {
      resolve(cb());
    }, timeout);
  });
  return p;
}

export const randomId = () => crypto.randomBytes(16).toString('hex');

export const mockPermissions = ({ isAuthorized = true } = {}) => {
  const injectAuthorized = (
    demandedPermissions,
    options,
    propName = 'isAuthorized'
  ) => withProps({ [propName]: isAuthorized });

  const branchOnPermissions = (
    demandedPermissions,
    FallbackComponent,
    options
  ) =>
    compose(
      injectAuthorized(demandedPermissions, options),
      branch(() => !isAuthorized, renderComponent(FallbackComponent))
    );
  return {
    injectAuthorized,
    branchOnPermissions,
    permissions: {},
  };
};

export const mockBranchOnFeatureToggle = ({
  isFeatureToggledOn = true,
} = {}) => (flags, FallbackComponent) =>
  branch(() => !isFeatureToggledOn, renderComponent(FallbackComponent));

export const mockComponent = displayName => {
  const MockedComponent = ({ children }) => children || null;
  MockedComponent.displayName = displayName;
  return MockedComponent;
};

export const mockConnector = (displayName, childrenProps = {}) => {
  const MockedConnector = jest.fn(({ children }) => children(childrenProps));
  MockedConnector.displayName = displayName;
  return MockedConnector;
};

export const mockContext = displayName => {
  let providedValue;

  const ProviderComponent = ({ value, children }) => {
    providedValue = value;
    return children;
  };
  ProviderComponent.displayName = `${displayName}.Provider`;
  const ConsumerComponent = ({ children }) => children(providedValue);
  ConsumerComponent.displayName = `${displayName}.Consumer`;

  return {
    Provider: ProviderComponent,
    Consumer: ConsumerComponent,
  };
};

export const mockReactIntl = () => ({
  ...require.requireActual('react-intl'),
  FormattedMessage: ({ id }) => `formatted-${id}`,
  injectIntl: withProps({ intl: intlMock }),
});

export const waitForPromises = () =>
  new Promise(resolve => setImmediate(resolve));
