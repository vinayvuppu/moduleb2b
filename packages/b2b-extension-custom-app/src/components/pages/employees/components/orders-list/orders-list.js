import PropTypes from 'prop-types';
import React from 'react';
import { injectIntl, FormattedMessage } from 'react-intl';
import flowRight from 'lodash.flowright';
import {
  LoadingSpinner,
  Spacings,
  Table,
  Constraints,
} from '@commercetools-frontend/ui-kit';
import { branchOnPermissions } from '@commercetools-frontend/permissions';

import FormattedDateTime from '@commercetools-local/core/components/formatted-date-time';
import { formatMoney } from '@commercetools-local/utils/formats/money';
import PageBottomSpacer from '@commercetools-local/core/components/page-bottom-spacer';
import TabContentLayout from '@commercetools-local/core/components/tab-content-layout';
import Pagination from '@commercetools-local/core/components/search/pagination';
import {
  orderStateMessages,
  paymentStateMessages,
  shipmentStateMessages,
} from '@commercetools-local/core/messages/order-states';
import { PERMISSIONS } from '../../../../../constants';

import OrdersListConnector from '../orders-list-connector';
import messages from './messages';

const columnDefinitions = [
  {
    key: 'orderNumber',
    label: <FormattedMessage {...messages.columnOrderNumber} />,
    flexGrow: 1,
  },
  {
    key: 'orderState',
    label: <FormattedMessage {...messages.columnOrderStatus} />,
    flexGrow: 1,
  },
  {
    key: 'paymentState',
    label: <FormattedMessage {...messages.columnPaymentState} />,
    flexGrow: 1,
  },
  {
    key: 'shipmentState',
    label: <FormattedMessage {...messages.columnShipmentState} />,
    flexGrow: 1,
  },
  {
    key: 'totalPrice',
    label: <FormattedMessage {...messages.columnOrderTotal} />,
    flexGrow: 1,
  },
  {
    key: 'createdAt',
    label: <FormattedMessage {...messages.columnCreatedAt} />,
    flexGrow: 1,
  },
];

export class OrdersList extends React.PureComponent {
  static displayName = 'OrdersList';

  static propTypes = {
    // regular props
    projectKey: PropTypes.string.isRequired,
    employeeId: PropTypes.string.isRequired,
    goToOrderDetails: PropTypes.func.isRequired,

    // HOC
    intl: PropTypes.shape({ formatMessage: PropTypes.func.isRequired }),
  };

  renderOrdersRow = ({ rowIndex, columnKey }, orders) => {
    const order = orders[rowIndex];
    let formattedValue = order[columnKey];

    if (columnKey === 'createdAt')
      formattedValue = (
        <FormattedDateTime type="datetime" value={formattedValue} />
      );
    else if (columnKey === 'totalPrice')
      formattedValue = formatMoney(formattedValue, this.props.intl);
    else if (columnKey === 'orderState')
      formattedValue = formattedValue ? (
        <FormattedMessage {...orderStateMessages[formattedValue]} />
      ) : (
        ''
      );
    else if (columnKey === 'paymentState')
      formattedValue = formattedValue ? (
        <FormattedMessage {...paymentStateMessages[formattedValue]} />
      ) : (
        ''
      );
    else if (columnKey === 'shipmentState')
      formattedValue = formattedValue ? (
        <FormattedMessage {...shipmentStateMessages[formattedValue]} />
      ) : (
        ''
      );

    return formattedValue;
  };

  handleOrderClick = orderId => this.props.goToOrderDetails(orderId);

  handleSortChange = (sortBy, sortDirection, searchQuery) => {
    searchQuery.set({
      ...searchQuery.value,
      page: 1,
      sorting: {
        key: sortBy,
        order: sortDirection.toLowerCase(),
      },
    });
  };

  render() {
    return (
      <OrdersListConnector
        employeeId={this.props.employeeId}
        projectKey={this.props.projectKey}
      >
        {({ ordersFetcher, searchQuery }) => {
          if (ordersFetcher.isLoading)
            return (
              <Spacings.Stack scale="m" alignItems="center">
                <LoadingSpinner>
                  <FormattedMessage {...messages.fetchingOrdersTitle} />
                </LoadingSpinner>
              </Spacings.Stack>
            );
          if (ordersFetcher.orders.results.length === 0)
            return <FormattedMessage {...messages.noOrdersTitle} />;

          return (
            <TabContentLayout
              data-testid="employee-details-orders-tab"
              data-track-component="EmployeesOrders"
            >
              <Constraints.Horizontal>
                <Table
                  columns={columnDefinitions}
                  rowCount={ordersFetcher.orders.results.length}
                  items={ordersFetcher.orders.results}
                  itemRenderer={rowData =>
                    this.renderOrdersRow(rowData, ordersFetcher.orders.results)
                  }
                  shouldFillRemainingVerticalSpace={true}
                  onSortChange={(sortBy, sortDirection) => {
                    this.handleSortChange(sortBy, sortDirection, searchQuery);
                  }}
                  sortBy={searchQuery.value.sorting.key}
                  sortDirection={searchQuery.value.sorting.order.toUpperCase()}
                  onRowClick={(e, index) =>
                    this.handleOrderClick(
                      ordersFetcher.orders.results[index].id
                    )
                  }
                >
                  <Pagination
                    totalItems={ordersFetcher.orders.total}
                    perPage={searchQuery.value.perPage}
                    page={searchQuery.value.page}
                    onPerPageChange={nextPerPage =>
                      searchQuery.set({
                        ...searchQuery.value,
                        perPage: nextPerPage,
                      })
                    }
                    onPageChange={nextPage =>
                      searchQuery.set({
                        ...searchQuery.value,
                        page: nextPage,
                      })
                    }
                  />
                </Table>
              </Constraints.Horizontal>
              <PageBottomSpacer />
            </TabContentLayout>
          );
        }}
      </OrdersListConnector>
    );
  }
}

export default flowRight(
  branchOnPermissions([PERMISSIONS.ViewOrders]),
  injectIntl
)(OrdersList);
