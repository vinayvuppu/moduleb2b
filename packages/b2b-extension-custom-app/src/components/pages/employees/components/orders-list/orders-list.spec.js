import PropTypes from 'prop-types';
import React from 'react';
import { shallow } from 'enzyme';
import { FormattedMessage } from 'react-intl';
import { LoadingSpinner } from '@commercetools-frontend/ui-kit';
import { intlMock } from '@commercetools-local/test-utils';
import FormattedDateTime from '@commercetools-local/core/components/formatted-date-time';
import OrdersListConnector from '../orders-list-connector';
import { OrdersList } from './orders-list';

jest.mock('@commercetools-local/utils/formats/date');

const createTestOrder = props => ({
  id: 'order-id-adf-asdf-adf',
  orderNumber: '123',
  orderState: 'Open',
  createdAt: '2016-11-15T16:41:27.167Z',
  totalPrice: {
    centAmount: '154619494463987360',
    currencyCode: 'GBP',
  },
  ...props,
});

const createTestProps = props => ({
  projectKey: 'test-project',
  employeeId: 'employee-id-foo-123',
  isActive: true,
  intl: intlMock,
  goToOrderDetails: jest.fn(),
  ...props,
});

const createConnectorProps = props => ({
  ordersFetcher: {
    isLoading: false,
    orders: {
      total: 1,
      count: 1,
      offset: 0,
      results: [createTestOrder()],
    },
  },
  searchQuery: {
    get: jest.fn(),
    set: jest.fn(),
    value: {
      page: 1,
      perPage: 20,
      sorting: {
        key: 'createdAt',
        order: 'desc',
      },
    },
  },
  ...props,
});

const TestRowItem = ({ children }) => <div>{children}</div>;
TestRowItem.propTypes = {
  children: PropTypes.any.isRequired,
};

describe('rendering', () => {
  let props;
  let wrapper;
  describe('when loading', () => {
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<OrdersList {...props} />);
      wrapper = wrapper.find(OrdersListConnector).renderProp('children')(
        createConnectorProps({
          ordersFetcher: {
            isLoading: true,
            orders: {
              total: 1,
              count: 1,
              offset: 0,
              results: [createTestOrder()],
            },
          },
        })
      );
    });

    it('should render the loading spinner', () => {
      expect(wrapper).toRender(LoadingSpinner);
    });
  });

  describe('without orders', () => {
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<OrdersList {...props} />);
      wrapper = wrapper.find(OrdersListConnector).renderProp('children')(
        createConnectorProps({
          ordersFetcher: {
            isLoading: true,
            orders: {
              total: 0,
              count: 0,
              offset: 0,
              results: [],
            },
          },
        })
      );
    });
    it('outputs correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should render base elements', () => {
      expect(wrapper).toRender(FormattedMessage);
    });
  });

  describe('with orders', () => {
    const orders = [createTestOrder()];
    let contentWrapper;
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<OrdersList {...props} />);
      contentWrapper = wrapper.find(OrdersListConnector).renderProp('children')(
        createConnectorProps()
      );
    });
    it('outputs correct tree', () => {
      expect(wrapper).toMatchSnapshot();
    });

    describe('rendering the table', () => {
      it('should render table with orders', () => {
        expect(
          contentWrapper
            .find('Table')
            .prop('columns')
            .map(_ => _.key)
        ).toEqual([
          'orderNumber',
          'orderState',
          'paymentState',
          'shipmentState',
          'totalPrice',
          'createdAt',
        ]);
      });
      describe('columns rendering', () => {
        const index = 0;
        it('should render column orderNumber', () => {
          const key = 'orderNumber';
          const wrapperRow = shallow(
            <TestRowItem>
              {wrapper
                .instance()
                .renderOrdersRow({ rowIndex: index, columnKey: key }, orders)}
            </TestRowItem>
          );
          expect(wrapperRow).toHaveText('123');
        });
        it('should render column orderState', () => {
          const key = 'orderState';
          const wrapperRow = shallow(
            <TestRowItem>
              {wrapper
                .instance()
                .renderOrdersRow({ rowIndex: index, columnKey: key }, orders)}
            </TestRowItem>
          );
          expect(wrapperRow).toRender({ id: 'OrderState.Open' });
        });
        it('should render column createdAt', () => {
          const key = 'createdAt';
          const wrapperRow = shallow(
            <TestRowItem>
              {wrapper
                .instance()
                .renderOrdersRow({ rowIndex: index, columnKey: key }, orders)}
            </TestRowItem>
          );
          expect(wrapperRow.find(FormattedDateTime)).toHaveProp(
            'value',
            '2016-11-15T16:41:27.167Z'
          );
        });
        it('should render column totalPrice', () => {
          const key = 'totalPrice';
          const wrapperRow = shallow(
            <TestRowItem>
              {wrapper
                .instance()
                .renderOrdersRow({ rowIndex: index, columnKey: key }, orders)}
            </TestRowItem>
          );
          expect(wrapperRow).toHaveText('GBP 1546194944639873.5');
        });
      });
    });
  });
  describe('callbacks', () => {
    describe('when a row is clicked', () => {
      beforeAll(() => {
        props = createTestProps();
        wrapper = shallow(<OrdersList {...props} />);
        wrapper = wrapper.find(OrdersListConnector).renderProp('children')(
          createConnectorProps()
        );
        wrapper.find('Table').prop('onRowClick')(null, 0);
      });

      it('should call `goToOrderDetails`', () => {
        expect(props.goToOrderDetails).toHaveBeenCalled();
      });
    });
  });
});
