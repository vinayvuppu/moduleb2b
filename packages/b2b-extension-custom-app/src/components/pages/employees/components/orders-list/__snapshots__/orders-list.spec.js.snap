// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with orders outputs correct tree 1`] = `
<withApplicationContext(withSearchViewRouterSearchQuery(Apollo(OrdersListConnector)))
  employeeId="employee-id-foo-123"
  projectKey="test-project"
>
  <Component />
</withApplicationContext(withSearchViewRouterSearchQuery(Apollo(OrdersListConnector)))>
`;

exports[`rendering without orders outputs correct tree 1`] = `
<Stack
  alignItems="center"
  scale="m"
>
  <LoadingSpinner
    scale="l"
  >
    <FormattedMessage
      defaultMessage="Loading orders..."
      description="Title shown while fetching orders."
      id="Employees.Details.OrdersForm.fetchingOrdersTitle"
      values={Object {}}
    />
  </LoadingSpinner>
</Stack>
`;
